pkg/bgpv1/manager/reconciler/service.go

```diff
@@ -298,31 +295,56 @@ func (r *LBServiceReconciler) svcDesiredRoutes(newc *v2alpha1api.CiliumBGPVirtua
 	if !svcSelector.Matches(serviceLabelSet(svc)) {
 		return nil, nil
 	}
-
-	// Ignore service managed by an unsupported LB class.
-	if svc.Spec.LoadBalancerClass != nil && *svc.Spec.LoadBalancerClass != v2alpha1api.BGPLoadBalancerClass {
-		// The service is managed by a different LB class.
-		return nil, nil
-	}
-
 	// Ignore externalTrafficPolicy == Local && no local endpoints.
 	if svc.Spec.ExternalTrafficPolicy == slim_corev1.ServiceExternalTrafficPolicyLocal &&
 		!hasLocalEndpoints(svc, ls) {
 		return nil, nil
 	}
-
 	var desiredRoutes []netip.Prefix
-	for _, ingress := range svc.Status.LoadBalancer.Ingress {
-		if ingress.IP == "" {
-			continue
-		}
-
-		addr, err := netip.ParseAddr(ingress.IP)
-		if err != nil {
-			continue
+	// Loop over the service advertisements and determine the desired routes.
+	for _, svcAdv := range newc.ServiceAdvertisements {
+		switch svcAdv {
+		case v2alpha1api.BGPLoadBalancerIPAddr:
+			if svc.Spec.Type != slim_corev1.ServiceTypeLoadBalancer {
+				continue
+			}
+			// Ignore service managed by an unsupported LB class.
+			if svc.Spec.LoadBalancerClass != nil && *svc.Spec.LoadBalancerClass != v2alpha1api.BGPLoadBalancerClass {
+				// The service is managed by a different LB class.
+				return nil, nil
```

Comment:
shouldn't we continue to next service advertisement, instead of returning here ?

Reply:
Updated.